core/src/Kokkos_View.hpp

```diff
@@ -449,6 +449,98 @@ struct ViewTraits {
 template <class DataType, class... Properties>
 class View;
 
+namespace Impl {
+
+/*
+ * \class ViewTracker
+ * \brief template class to wrap the shared allocation tracker
+ *
+ * \section This class is templated on the View and provides
+ * constructors that match the view.  The constructors and assignments
+ * from view will externalize the logic needed to enable/disable
+ * ref counting to provide a single gate to enable further developments
+ * which may hing on the same logic.
+ *
+ */
+template <class ParentView>
+struct ViewTracker {
+  typedef Kokkos::Impl::SharedAllocationTracker track_type;
+  typedef typename ParentView::traits view_traits;
```

Comment:
`using`, not `typedef` in new code, please

Reply:
changed